pytest_services/locks.py

```diff
@@ -80,23 +82,39 @@ def unlock_display(display, lock_dir, services_log):
     return unlock_resource('display', display, lock_dir, services_log)
 
 
-def lock_resource(name, resource_getter, lock_dir, services_log):
-    """Issue a lock for given resource."""
-    with locked_resources(name, lock_dir) as bound_resources:
-        services_log.debug('bound_resources {0}: {1}'.format(name, bound_resources))
-
-        resource = resource_getter(bound_resources)
-        while resource in bound_resources:
-            # resource is already taken by someone, retry
-            services_log.debug('bound resources {0}: {1}'.format(name, bound_resources))
-            resource = resource_getter(bound_resources)
-        services_log.debug('free resource choosen {0}: {1}'.format(name, resource))
-        bound_resources.append(resource)
-        services_log.debug('bound resources {0}: {1}'.format(name, bound_resources))
-        return resource
+@pytest.fixture(scope='session')
+def lock_resource_timeout():
+    """Max number of seconds to lock resource."""
```

Comment:
this is the timeout for obtaining the lock, for the lock time (as it's not defined at all)
also, I'd keep set it by default to socket.getdefaulttimeout()

Reply:
I changed to 
```
@pytest.fixture(scope='session')
def obtain_lock_resource_timeout():
    """Timeout for obtaining resource lock."""
    return socket.getdefaulttimeout()
```
but `socket.getdefaulttimeout()` gives None. How come?

Is the rename to `obtain_lock_resource_timeout` ok?

Reply:
the name of the fixture was good enough, my comment was only about the docstring
about the getdefaulttimeout - sorry my bad, I thought it'd return 20sec, so I'd have it just 20 sec then as a default

Reply:
No problem. Hmm, 20 sec is a long time to wait. In my case I expect it to be done in a fraction of a second but there might be some scenario I'm unfamiliar with. I'll increase it to 20s.